src/d3a/models/market/two_sided.py

```diff
@@ -264,9 +278,36 @@ def match_recommendation(self, recommended_list):
                     recommended_list, index+1, trade, bid_trade
                 )
 
+    @staticmethod
+    def validate_requirements_satisfied(offer: Offer, bid: Bid) -> None:
+        """Validate if both trade parties satisfy each other's requirements.
+
+        :raises:
+            InvalidBidOfferPair: Bid offer pair failed the validation
+        """
+        if ((offer.requirements or bid.requirements) and
```

Comment:
You mean
```suggestion
        if ( not (offer.requirements or bid.requirements) or
```
Right?

Reply:
Not really, actually I'm checking if at least the bid or offer have requirements, if True then we'll move to the second check which is to see if these requirements are satisfied
So if both offer and bid have no requirements, the statement will be False and it won't raise an exception

Reply:
Yes, you are totally right!